.github/workflows/frontend-check.yaml

```diff
@@ -25,6 +25,36 @@ jobs:
       - name: run component tests
         run: podman run --rm -it e2e-frontend run cypress:cp
 
+  test-and-coverage:
+    name: Test and Generate Coverage
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: ./frontend
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18' 
+          cache: 'npm'
+          cache-dependency-path: ./package-lock.json
+
+      - name: Install dependencies
+        run: npm ci
+      - name: Run Cypress tests and generate coverage
+        run: npm run cypress:run:ci
```

Comment:
It's somewhat interesting that we seem to be generating coverage data based on the functional tests (e2e) rather than on the unit tests as we do on the backend. That's not necessarily bad (I'd kinda like to do both)... but we're already running the frontend functional tests under the `frontend/tests/e2e.sh` script -- doing it twice to collect coverage data seems inefficient. Can we gather the coverage data under `e2e.sh`? 

And does the `cypress:run:ci` script really work? The functional tests require a backend and databases, and it's not clear how all that works in this context. 😕 

(Or maybe it doesn't, and that's the problem we're seeing in the CI run ... ??)